include/boost/url/impl/url_base.ipp

```diff
@@ -1650,21 +1654,22 @@ parse() noexcept
             detail::id_query] &&
             *p == '&' ));
     auto p0 = p++;
-    auto const ek =
-        detail::qkey_pct_set();
-    error_code ec;
-    p = ek.parse(p, end, ec);
-    BOOST_ASSERT(! ec);
-    nk_ = p - p0;
-    if(p == end)
+    if(p != end && *p != '=' && *p != '&')
```

Comment:
it helps coverage to break these things up on individual lines:
```
if( p != end &&
    *p != '=' &&
    *p != '&')
```

This way we know that each condition is covered